velodyne_pointcloud/include/velodyne_pointcloud/point_types.h

```diff
@@ -0,0 +1,48 @@
+/* -*- mode: C++ -*-
+ *
+ *  Copyright (C) 2011, 2012 Austin Robot Technology
+ *
+ *  License: Modified BSD Software License Agreement
+ *
+ *  $Id: data_base.h 1554 2011-06-14 22:11:17Z jack.oquin $
+ */
+
+/** \file
+ *
+ *  Point Cloud Library point structures for Velodyne data.
+ *
+ *  @author Jesse Vera
+ *  @author Jack O'Quin
+ *  @author Piyush Khandelwal
+ */
+
+#ifndef VELODYNE_POINTCLOUD_POINT_TYPES_H
+#define VELODYNE_POINTCLOUD_POINT_TYPES_H
+
+#include <pcl/point_types.h>
+
+namespace velodyne_pointcloud
+{
+/** Euclidean Velodyne coordinate, including intensity and ring number. */
+struct PointXYZIR
+{
+  float x;                         ///< x coord
+  float y;                         ///< y coord
+  float z;                         ///< z coord
```

Comment:
Just to make absolutely sure there aren't any problems, please use the PCL macro `PCL_ADD_POINT4D;` instead of these 3 types.

Reply:
Then we would need to add a padding (It adds four folats) to the point cloud, which increases the size of the buffer and cloud object. This would result in adding another container with a padding (XYZPIR) as default. Your choice. ;)

Reply:
@JWhitleyWork What would you prefer? adding a useless padding to the could within a new container?

Reply:
PointXYZIR with the quad-word XYZ was the original version. Even if it had additional padding, it's what was there before so we should use it as the default and keep it the same as the original name to avoid compatibility issues. Here are the contents of this file from the released version, 1.5.2:

```
namespace velodyne_pointcloud
{
  /** Euclidean Velodyne coordinate, including intensity and ring number. */
  struct PointXYZIR
  {
    PCL_ADD_POINT4D;                    // quad-word XYZ
    float    intensity;                 ///< laser intensity reading
    uint16_t ring;                      ///< laser ring number
    EIGEN_MAKE_ALIGNED_OPERATOR_NEW     // ensure proper alignment
  } EIGEN_ALIGN16;

}; // namespace velodyne_pointcloud


POINT_CLOUD_REGISTER_POINT_STRUCT(velodyne_pointcloud::PointXYZIR,
                                  (float, x, x)
                                  (float, y, y)
                                  (float, z, z)
                                  (float, intensity, intensity)
                                  (uint16_t, ring, ring))
```